protos/mocap/mocap.proto

```diff
@@ -0,0 +1,164 @@
+syntax = "proto3";
+
+package mavsdk.rpc.mocap;
+
+option java_package = "io.mavsdk.mocap";
+option java_outer_classname = "MocapProto";
+
+/**
+ * Motion Capture allow vehicles to navigate when a global
+ * position source is unavailable or unreliable
+ * (e.g. indoors, or when flying under a bridge. etc.).
+ */
+service MocapService {
+    // Send Global position/attitude estimate from a vision source.
+    rpc SetVisionPositionEstimate(SetVisionPositionEstimateRequest) returns(SetVisionPositionEstimateResponse) {}
+    // Send motion capture attitude and position.
+    rpc SetAttitudePositionMocap(SetAttitudePositionMocapRequest) returns(SetAttitudePositionMocapResponse) {}
+    // Send odometry information with an external interface.
+    rpc SetOdometry(SetOdometryRequest) returns(SetOdometryResponse) {}
+}
+
+message SetVisionPositionEstimateRequest {
+    VisionPositionEstimate vision_position_estimate = 1;
+}
+message SetVisionPositionEstimateResponse {
+    MocapResult mocap_result = 1;
+}
+
+message SetAttitudePositionMocapRequest {
+    AttitudePositionMocap attitude_position_mocap = 1;
+}
+message SetAttitudePositionMocapResponse {
+    MocapResult mocap_result = 1;
+}
+
+message SetOdometryRequest {
+    Odometry odometry = 1;
+}
+message SetOdometryResponse {
+    MocapResult mocap_result = 1;
+}
+
+// Global position/attitude estimate from a vision source.
+message VisionPositionEstimate {
+    uint64 time_usec = 1; // PositionBody frame timestamp UNIX Epoch time (0 to use Backend timestamp)
+    PositionBody position_body = 2; // Global position (m)
+    AngleBody angle_body = 3; // Body angle (rad).
+    Covariance pose_covariance = 4; // Pose cross-covariance matrix.
+}
+
+// Motion capture attitude and position
+message AttitudePositionMocap {
+    uint64 time_usec = 1; // PositionBody frame timestamp UNIX Epoch time (0 to use Backend timestamp)
+    Quaternion q = 2; // Attitude quaternion (w, x, y, z order, zero-rotation is 1, 0, 0, 0)
+    PositionBody position_body = 3; // Body Position (NED)
+    Covariance pose_covariance = 4; // Pose cross-covariance matrix.
+}
+
+// Odometry message to communicate odometry information with an external interface.
+message Odometry {
+    // Mavlink frame id
+    enum MavFrame {
+        STUB = 0; // Stub for Protobuf
```

Comment:
Why do we need a `STUB` for protobuf?

Reply:
```bash
$ ./generate_from_protos.sh 
-------------------------------
 Generating pb and grpc.pb files
    * protoc --version: libprotoc 3.9.1
-------------------------------

mocap/mocap.proto:64:20: The first enum value must be zero in proto3.
```

Reply:
Doesn't that mean that `BODY_FRD` should be `= 0` instead of `= 12`? See here: https://developers.google.com/protocol-buffers/docs/proto3#enum.

And then the next ones like `BODY_FLU` could be `1`, `2`, etc. Or does it conflict with something else?

Reply:
I think that setting `BODY_FRD = 0` may be confusing for gRPC-client users.
For example in MAVSDK-Python:
I [manually changed generated code](https://github.com/mavlink/MAVSDK-Python/pull/111/files#diff-755143e2465cb1cce8758a454a7ccdd4R273-R274) to delete `STUB` and set real values of `enum` members.
This will not conflict for any user-defined code which operated with MAVLink-defined constants  (for example `MOCAP_NED` eq `14`).
And also, on the other hand, bindings for gRPC plugins to other languages will be produced with users which familiar with MAVSDK and MAVLink and they will can also remove stubs from final API.
What

Reply:
How is `BODY_FRD = 0` confusing? I don't get that :confused:.

Reply:
Because having to change all the templates for all the languages just to say "ignore any enum value called `STUB` with a value of "0" because that's a workaround we did" is quite a strong and non-intuitive thing. I'd like to be sure we need that before :sweat_smile:.

Reply:
I mean that if some constant values like `BODY_FRD` will be differ from its values in MAVLink then it can seem like collision.
For example if in MAVSDK-Proto `BODY_FRD = 0` so in MAVSDK-Python (and other bindings) this constant also eq `0` as default, but in MAVLink `BODY_FRD` = `12`. 
User code (in python for example) may check if current frame is `Mocap.BODY_FRD` and expect `12`, but this constant eq `0`

Reply:
> Because having to change all the templates for all the languages just to say "ignore any enum value called `STUB` with a value of "0" because that's a workaround we did" is quite a strong and non-intuitive thing. I'd like to be sure we need that before .


Yes, I think that this stub (:smile:) not a best solution.
But other way (dont use `STUB`) looks more potentially dangerous for me. 
I do not insist on this decision, but I think that in case of dont use `STUB` maintainers should manually change values in bindings like MAVSDK-Python.
And also `static_cast` will not work for casting ever more in `plugin_service_impl.h` :)

Reply:
I don't think I agree on the fact that this is a reason for adding a very weird condition in all the generated code :sweat_smile:. This is adding confusion in the templates in all languages, and they are already not super easy to read. And on the other side, I don't see it as a good practice to rely on the id of an enum value.

For instance, I don't think that protobuf says in the specs that `BODY_FRD = 0` will always translate that ID in the resulting generated code. Maybe it works now in Python, but that may well be out of luck. Or is that somewhere in the specs?

Reply:
> And also static_cast will not work for casting ever more in plugin_service_impl.h :)

Yes, `static_cast` is convenient, but it requires a unit test to make sure that it works. If it doesn't, then we need to write more verbose code, which I believe can still be auto-generated.

Reply:
Few more info from [docs](https://developers.google.com/protocol-buffers/docs/proto3#default):

> For enums, the default value is the first defined enum value, which must be 0.

So `0` is default. I don't know in which cases uses default values. May be this also need to take into consideration.

Reply:
That's the default value in case you don't specify it. In many cases we have `UNKNOWN = 0`. But also, we always specify a value in the code we generate. So I don't think that has any effect for us.

Reply:
I would vote for this: we remove `STUB`, and we start the others at 0 (i.e. we don't care about the MAVLink enum ids). If then we see that this makes it very confusing for python users, we can change the python templates to make it more intuitive.

This way we don't propagate a weird `STUB` to all the other languages.

What do you think? :blush:

Reply:
I think that this make sense. I will do corresponding changes.